src/lib.rs

```diff
@@ -154,26 +154,19 @@ fn vec_vtable<S: 'static + New + GetVal>() -> Vec<*mut ()> {
     assert_eq!(size_of::<Box<()>>(), size_of::<usize>());
     assert_eq!(size_of::<Box<dyn GetVal>>(), size_of::<usize>() * 2);
     let mut v = Vec::with_capacity(*VEC_SIZE);
-    // Since every instance of S will share the same vtable, it's OK for us to pull it out and
-    // reuse it. With the coerce_unsized feature turned on, we can do this in a marginally cleverer
-    // way, but the outcome is the same.
-    let vtable = {
-        let s = S::new();
-        let b: &dyn GetVal = &s;
-        let (_, vtable) = unsafe { transmute::<_, (usize, usize)>(b) };
-        vtable
-    };
     // We're going to lay out memory (on a 64-bit machine) as:
     //   offset 0: vtable
     //          8: S
     let (layout, _) = Layout::new::<usize>().extend(Layout::new::<S>()).unwrap();
     // The following assert ensure that the layout really is as we expect.
     assert_eq!(layout.size(), size_of::<usize>() + size_of::<S>());
     for _ in 0..*VEC_SIZE {
+        let s = S::new();
         let b = unsafe {
+            let (_, vtable) = transmute::<&dyn GetVal, (usize, usize)>(&s);
```

Comment:
So you suck it straight out of the trait type instead of an instance?

Reply:
Not really. Basically if you have an `&S` reference but say you want an `&dyn GetVal` reference, Rust will automatically coerce the former into the latter. So I'm just making use of that feature here: the function's argument will be automatically coerced into a fat pointer because of the type annotation.